src/createServer.js

```diff
@@ -1,7 +1,356 @@
 'use strict';
 
+const express = require('express');
+const { Category } = require('./models/Category.model');
+const { Expense } = require('./models/Expense.model');
+const { User } = require('./models/User.model');
+const router = express.Router();
+const { Op } = require('sequelize');
+
+router.get('/categories', async (req, res) => {
+  try {
+    const categories = await Category.findAll();
+
+    res.json(categories);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error fetching categories', error: error.message });
+  }
+});
+
+router.get('/categories/:id', async (req, res) => {
+  try {
+    const category = await Category.findByPk(req.params.id);
+
+    if (!category) {
+      return res.status(404).json({ message: 'Category not found' });
+    }
+
+    res.json(category);
+  } catch (error) {
+    res
+      .status(500)
+      .json({ message: 'Error fetching category', error: error.message });
+  }
+});
+
+router.post('/categories', async (req, res) => {
+  try {
+    const { name, description } = req.body;
+
+    if (!name) {
+      return res.status(400).json({ message: 'Category name is required' });
+    }
+
+    const newCategory = await Category.create({
+      name,
+      description: description || null,
+    });
+
+    res.status(201).json(newCategory);
+  } catch (error) {
+    res
+      .status(400)
```

Comment:
Consider using a 500 status code for server errors to maintain consistency with other error responses in the code.